### PR TITLE
return os.ErrNotExist in some cases for empty TLF

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/kbfs/libfs/fs.go [hunk 1]
```diff
@@ -569,8 +569,10 @@ func (fs *FS) ensureParentDir(filename string) error {
 	return nil
 }
 
-func (fs *FS) requireNonEmpty() error {
-	if fs.empty {
+func (fs *FS) checkEmpty(notExistErrIfEmpty bool) error {
```
Comment (jzila) [inline go/kbfs/libfs/fs.go]: I'd just call this field `errIfEmpty`.

File: go/kbfs/libfs/fs.go [hunk 2]
```diff
@@ -691,7 +693,7 @@ func (fs *FS) Rename(oldpath, newpath string) (err error) {
 		err = translateErr(err)
 	}()
 
-	if err := fs.requireNonEmpty(); err != nil {
+	if err := fs.checkEmpty(false); err != nil {
```
Comment (jzila) [inline go/kbfs/libfs/fs.go]: Why `false` here? Shouldn't we error if the fs is empty?

Comment (jakob223) [inline go/kbfs/libfs/fs.go]: Can you make this an enum or make these calls include a comment as to what this boolean parameter means (e.g. `checkEmpty(true /*notExistErrIfEmpty*/)`

Comment (songgao) [reply go/kbfs/libfs/fs.go]: We still error, but get `os.ErrNotExist`, which is consistent with when you try to `mv non-existing-file some-other-name` in shell. But I can see this returning the other error too. Not really sure how to decide.

File: go/kbfs/libfs/fs.go (hunk 1 above)
Comment (jakob223) [reply go/kbfs/libfs/fs.go]: it chooses which error to return, not whether to return one at all.

**Comment by songgao** [comment]
@jakob223 @jzila I renamed the function, added enum, and added some comments. PTAL when you have time. Thanks!

**Pull request merged**